lib/internal/modules/cjs/loader.js

```diff
@@ -567,7 +570,16 @@ function tryModuleLoad(module, filename) {
 }
 
 Module._resolveFilename = function(request, parent, isMain, options) {
-  if (NativeModule.nonInternalExists(request)) {
+  if (request.startsWith('@nodejs/')) {
+    request = request.slice(8);
+    if (!namespaceWhitelist.has(request)) {
+      throw new ERR_INVALID_NODE_BUILTIN(request);
+    }
+    return request;
+  }
+
+  if (!nonNamespaceBlacklist.has(request) &&
+      NativeModule.nonInternalExists(request)) {
```

Comment:
Need to benchmark this change

Reply:
Do you have a suggestion on how to do this or an exiting benchmark we can run?

Reply:
It's possible a new benchmark would need to be written. We need to know the impact here